File: src/atdd/coder/validators/fixtures/silent_swallow/typescript_clean/observedHandlers.ts
// Fixture: acceptable TS shapes.

interface Match {
  id: string;
}

declare const logger: {
  warn: (msg: string, ctx?: unknown) => void;
  info: (msg: string, ctx?: unknown) => void;
};

declare function createMatch(playerId: string): Match;
declare function remoteLookup(playerId: string): string;
declare function cachedValue(playerId: string): string;

export function reThrowAfterLog(playerId: string): string {
  try {
    return createMatch(playerId).id;
  } catch (e) {
    logger.warn("match_creator failed", { playerId, error: e });
    throw e;
  }
}

export function fallbackAfterLog(playerId: string): string {
  try {
    return remoteLookup(playerId);
  } catch (e) {
    logger.warn("remote lookup failed, using fallback", { playerId, error: e });
    return cachedValue(playerId);
  }
}

export function rethrowWithContext(playerId: string): string {
  try {
    return createMatch(playerId).id;
  } catch (e) {
    throw new Error(`match creation failed for ${playerId}: ${e}`);
  }
}

export function suppressedSwallow(playerId: string): string {
  try {
    return createMatch(playerId).id;
  } catch (e) { // atdd:suppress(coder.logging.coach-silent-swallow)
    return "";
  }
}

File: src/atdd/coder/validators/fixtures/silent_swallow/python_clean/observed_handlers.py
"""Fixture: acceptable shapes — handlers that log + re-raise or log + fallback."""

import logging

logger = logging.getLogger(__name__)


def re_raise_after_log(player_id: str) -> str:
    try:
        return _create_match(player_id).id
    except Exception as e:
        logger.warning(
            "match_creator failed",
            extra={"player_id": player_id, "error": str(e)},
        )
        raise


def fallback_after_log(player_id: str) -> str:
    try:
        return _remote_lookup(player_id)
    except Exception as e:
        logger.warning(
            "remote lookup failed, using fallback",
            extra={"player_id": player_id, "error": str(e)},
        )
        return _cached_value(player_id)


def reraise_with_context(player_id: str) -> str:
    try:
        return _create_match(player_id).id
    except ValueError as e:
        raise RuntimeError(f"match creation failed for {player_id}") from e


def returns_none_naturally(player_id: str) -> str | None:
    try:
        return _maybe_lookup(player_id)
    except KeyError as e:
        logger.info("lookup miss", extra={"player_id": player_id, "error": str(e)})
        return None


def suppression_pragma(player_id: str) -> str:
    try:
        return _create_match(player_id).id
    except Exception:  # atdd:suppress(coder.logging.coach-silent-swallow)
        return ""


class _Match:
    id: str = ""


def _create_match(player_id: str) -> _Match:
    return _Match()


def _remote_lookup(player_id: str) -> str:
    return player_id


def _cached_value(player_id: str) -> str:
    return player_id


def _maybe_lookup(player_id: str) -> str:
    return player_id
